kebechet_administrator.py

```diff
@@ -0,0 +1,148 @@
+#!/usr/bin/env python3
+# workflow-helpers Kebechet Administrator
+# Copyright(C) 2020 Sai Sankar Gochhayat
+#
+# This program is free software: you can redistribute it and / or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program. If not, see <http://www.gnu.org/licenses/>.
+
+"""Kebechet Administrator.
+
+This script run in a workflow task to take and incoming message and decide and decides which repositores
+Kebechet needs to be run on and schedules the necessary messages.
+"""
+
+import logging
+import json
+from typing import Dict
+from thoth.storages import GraphDatabase
+from thoth.workflow_helpers.configuration import Configuration
+from thoth.messaging import __all__ as all_messages
+from thoth.workflow_helpers import __service_version__
+
+__COMPONENT_NAME__ = "Kebechet Administrator"
+
+_LOGGER = logging.getLogger("thoth.run_kebechet_administrator")
+_LOGGER.info("Thoth workflow-helpers task: run_kebechet_administrator v%s", __service_version__)
+
+# Estabilish connection to Database.
+GRAPH = GraphDatabase()
+GRAPH.connect()
+
+_URL_PREFIX = "https://github.com/"
+
+output_messages = []  # Messages to be sent by producer.
+
+
+def _handle_solved_message(Configuration):  # noqa: N803
+    """Handle all the messages for which Kebechet needs to run on if the sovler type matches the os type."""
+    solver_string = Configuration.get("THOTH_SOLVER_NAME")  # ex - solver-fedora-31-py38
+    _, os_name, os_version, python_version = solver_string.rsplit(sep="-", maxsplit=3)
+    python_version = ".".join([i for i in python_version if i.isdigit()])  # generates '3.8' from 'py39'
+    repositories: Dict[str, Dict] = GRAPH.get_kebechet_github_installations_info_for_python_package_version(
+        package_name=Configuration.PACKAGE_NAME,
+        index_url=Configuration.PACKAGE_INDEX,
+        os_name=os_name,
+        os_version=os_version,
+        python_version=python_version,
+    )  # We query without the package_version.
+    for key in repositories.keys():
+        repo_info = repositories[key]
+        # Construct the message input
+        if repo_info.get("private"):
+            continue  # We skip for private repo's.
+        if repo_info.get("package_version") == Configuration.PACKAGE_VERSION:
```

Comment:
@pacospace @fridex  Should this be a lower than this version comparison or something? Would the package version from the solved message be like a string "1.2.0"? Should we use a version parser to compare these strings?

Reply:
Yes I think it is required, as in the message is a string: https://github.com/thoth-station/messaging/blob/5cf33f36a93525ab739d57c2d46d8b860f9e2adb/thoth/messaging/solved_package.py#L38

Reply:
I have added a semver compare to schedule if the user package version is lower than solved package version